### PR TITLE
Update url.py to fit WSL path that starts with "\\wsl.local"

### PR DESCRIPTION
**Pull request by 95833**

### CODE REVIEW

File: plugin/core/url.py [hunk 1]
```diff
@@ -56,7 +56,8 @@ def parse_uri(uri: str) -> tuple[str, str]:
         path = url2pathname(parsed.path)
         if os.name == 'nt':
             netloc = url2pathname(parsed.netloc)
-            path = path.lstrip("\\")
+            if path[0] == '\\':
```
Comment (jwortmann) [inline plugin/core/url.py:59]:
```suggestion
            if path.startswith('\\'):
```

**Comment by jwortmann** [comment]
Could you add a test in the `tests/test_url.py` file with an example, i.e. some URL that is not handled correctly at the moment but would be fixed with this pull request?

I see there is already a test for a WSL URL at https://github.com/sublimelsp/LSP/blob/c64ac00a1295fdc8461811033e0974443fd9ea55/tests/test_url.py#L39-L42
but you could add another one after that which would be handled by the change in this pull request.

**Comment by 95833** [comment]
Can you teach me how to use tests?

**Comment by 95833** [comment]
I don't know how to run this test.

**Comment by jwortmann** [comment]
The CI check on GitHub will run it after you pushed another commit.

**Comment by 95833** [comment]
Can't I run it locally? It's a bit annoying to have to push to GitHub every time I test the code.

**Comment by jwortmann** [comment]
To manually run tests locally you need to install the UnitTesting package from package control. See https://github.com/sublimelsp/LSP/blob/main/CONTRIBUTING.md#testing

I haven't done this though, but usually you know what the expected outcome should be, so it should be fine to just push new commits until it works. The commits will be squashed into a single one when the PR gets merged anyway.

**Comment by 95833** [comment]
Okay

**Comment by 95833** [comment]
To provide support for WSL paths, I think it would be better to update the underlying packages related to path handling rather than changing the program logic; the issue seems to have become somewhat complex.

**Pull request closed without merging**